src/gui/MainWindow.cpp

```diff
@@ -282,15 +284,44 @@ void MainWindow::initXournalWidget() {
     scrollHandling->init(this->xournal->getWidget(), layout);
 }
 
-void MainWindow::setTouchscreenScrollingForDeviceMapping() {
+void MainWindow::setGtkTouchscreenScrollingForDeviceMapping() {
     for (InputDevice const& inputDevice: DeviceListHelper::getDeviceList(this->getControl()->getSettings())) {
         InputDeviceClass deviceClass = InputEvents::translateDeviceType(inputDevice.getName(), inputDevice.getSource(),
                                                                         this->getControl()->getSettings());
         if (inputDevice.getSource() == GDK_SOURCE_TOUCHSCREEN && deviceClass != INPUT_DEVICE_TOUCHSCREEN) {
-            gtk_scrolled_window_set_kinetic_scrolling(GTK_SCROLLED_WINDOW(winXournal), false);
-            break;
+            setGtkTouchscreenScrollingEnabled(false);
+
+            return;
         }
     }
+
+    setGtkTouchscreenScrollingEnabled(true);
+}
+
+void MainWindow::setGtkTouchscreenScrollingEnabled(bool enabled) {
+    if (enabled == gtkTouchscreenScrollingEnabled || usingTouchWorkaround || winXournal == nullptr) {
+        return;
+    }
+
+    gtkTouchscreenScrollingEnabled = enabled;
+
+    // Run on the main thread:
+    // 0 = timeout (run as soon as possible)
+    gdk_threads_add_timeout(
```

Comment:
**nitpick (non-blocking):** use `gdk_threads_add_idle*` here instead of `gdk_threads_add_timeout`

Reply:
I tried `gtk_threads_add_idle(`, but the priority it put calling the function at too low of a priority!

I'm looking into using `gtk_threads_add_idle_full` with a priority of `G_PRIORITY_HIGH_IDLE`!

Reply:
According to the GTK documentation, `gtk_threads_add_idle_full` has the following signature:
```
guint
gdk_threads_add_idle_full (gint priority,
                           GSourceFunc function,
                           gpointer data,
                           GDestroyNotify notify);
```

Personally, `gdk_threads_add_timeout` seems more readable here; however, I understand that this might not be a question of readability.

Which do you think is more readable/better?

Reply:
It is a question of understandability and this is a subset of readability. `gtk_threads_add_idle` looks more like adding a task to the scheduler. A timeout looks like, that the code must be executed after a specific amount of time with a specific frequency. But also with your above comment this should be the same.
I leave that to you and @LittleHuba.

Regarding my opinion: `gtk_threads_add_idle` is more readable, since it also tells the reader which priority must be used. Also, the `GDestroyNotify` can be a `nullptr` as I know. This would only add one parameter and will replace the magic number `0` with a named constant.

Reply:
I found `Util::execInUiThread`! I'm adding a `priority` parameter and using that!

Reply:
`Util::execInUiThread` is actually meant for exactly this usecase.

Reply:
Changed!